server/src/controllers/apiController.ts

```diff
@@ -1,26 +1,33 @@
 import { Request, Response } from 'express';
 import { GptService } from '../services/gptService';
+// import fs from 'fs';
 
 export const ApiController = {
   async process(req: Request, res: Response) {
     try {
-      if (!req.file) {
-        return res.status(400).json({ success: false, message: 'No file uploaded' });
-      }
-
-      if (req.file.mimetype !== 'image/jpeg') {
-        return res.status(400).json({
-          success: false,
-          message: `Must upload a JPEG image. Your image is minetype: ${req.file.mimetype}`,
-        });
+      if (!req.file || !req.file?.buffer) {
+        return res.status(400).json({ success: false, message: 'Error, no file or no buffer!' });
       }
 
       console.log('successfully saved the image - creating gpt request');
 
-      let gptService = new GptService();
-      let data = gptService.doSomething();
+      // Buffer -> base64
+      const encodedImg = req.file.buffer.toString('base64');
+      // DEBUG: if you want to check to make sure that the conversion works properly
+      // fs.writeFile('debug_base64.txt', encodedImg, (err) => {
+      //   if (err) {
+      //     console.error('Error writing Base64 string to file:', err);
+      //   } else {
+      //     console.log('Base64 string saved to debug_base64.txt');
+      //   }
+      // });
+
+      const gptService = new GptService();
+      const data = await gptService.doSomething(encodedImg);
```

Comment:
instead of calling doSomething let's all the GPT service you wrote,

```suggestion
      const data = await gptService.sendToGpt(encodedImg);
```